base/reduce.jl

```diff
@@ -256,8 +256,35 @@ foldr(op, itr; kw...) = mapfoldr(identity, op, itr; kw...)
     end
 end
 
-mapreduce_impl(f, op, A::AbstractArrayOrBroadcasted, ifirst::Integer, ilast::Integer) =
-    mapreduce_impl(f, op, A, ifirst, ilast, pairwise_blocksize(f, op))
+# vararg version
+@noinline function mapreduce_impl(f::F, op, A::AbstractArrayOrBroadcasted,
+                                  ifirst::Integer, ilast::Integer, blksize::Int,
+                                  Bs::Vararg{AbstractArrayOrBroadcasted,N}) where {F,N}
+    Xs = (A, Bs...)
+    if ifirst == ilast
+        @inbounds x1 = ith_all(ifirst, Xs)
```

Comment:
This could be combined with the original version? It is a bit of an ugly signature, `mapreduce_impl(f, op, A, i,j,b, Bs...)`; not sure how hard it would be to change that to put all the arrays last.